daml-script/dump/src/main/scala/com/daml/script/dump/TreeUtils.scala

```diff
@@ -214,25 +218,33 @@ object TreeUtils {
 
   object SimpleCommand {
     def fromCommand(command: Command, tree: TransactionTree): Option[SimpleCommand] = {
+      def simpleExercise(exercisedEvent: ExercisedEvent): Option[ContractId] = {
+        val result = exercisedEvent.exerciseResult.flatMap {
+          _.sum match {
+            case Sum.ContractId(value) => Some(value)
+            case _ => None
+          }
+        }
+        val creates = treeEventCreatedCids(Kind.Exercised(exercisedEvent), tree).toSeq
+        (result, creates) match {
+          case (Some(cid), Seq(createdCid)) if cid == createdCid =>
+            Some(ContractId(cid))
+          case _ => None
+        }
+      }
       command match {
         case CreateCommand(createdEvent) =>
           Some(SimpleCommand(command, ContractId(createdEvent.contractId)))
         case ExerciseCommand(exercisedEvent) =>
-          val result = exercisedEvent.exerciseResult.flatMap {
-            _.sum match {
-              case Sum.ContractId(value) => Some(value)
-              case _ => None
-            }
-          }
-          val creates = treeEventCreatedCids(Kind.Exercised(exercisedEvent), tree)
-          (result, creates) match {
-            case (Some(cid), Seq(createdCid)) if cid == createdCid =>
-              Some(SimpleCommand(command, ContractId(cid)))
-            case _ => None
+          simpleExercise(exercisedEvent).map(SimpleCommand(command, _))
+        case CreateAndExerciseCommand(_, exercisedEvent) =>
+          if (exercisedEvent.consuming) {
+            // If the choice is not consuming then we have two resulting contracts:
+            // The created one and the result of the exercise. I.e. not a simple command.
```

Comment:
Non-consuming choices can still call consuming choices (e.g., `Archive`). Totally fine not to handle this case but perhaps worth mentioning in the comment.

Reply:
True, I've expanded the comment. It shouldn't be too hard to expand this to take nested consuming choices into account. I'll look into it separately, though.

Reply:
Addressed in https://github.com/digital-asset/daml/pull/9185